### PR TITLE
Support correlated scalar aggregation subqueries (v8)

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
Supersedes #6190

**Comment by martint** [comment]
I merged everything but the docs.

@electrum, can you review the documentation changes?

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 1]
```diff
@@ -732,3 +732,52 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+Scalar subquery
+^^^^^^^^^^^^^^^
+
+.. code-block:: sql
+
+    SELECT nation.name, region.name
+    FROM nation n
+    WHERE regionkey = (SELECT max(regionkey) FROM region);
+
+The result of scalar subquery is a row returned arbitrary ``SELECT`` statement. If the query returns more than single row then runtime error is propagated, in case of no rows then ``null`` is returned.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/sql/select.rst]:
```
A scalar subquery is a non-correlated subquery that returns zero or
one row. It is an error for the subquery to produce more than one
row. The returned value is ``NULL`` if the subquery produces no rows::
```
Same comment about moving example query below

**Comment by electrum** [comment]
Please remove the additional non-documentation commits from this PR.

**Comment by kokosing** [comment]
@electrum Rebased and comments addressed. Superseded by https://github.com/prestodb/presto/pull/6228

**Pull request closed without merging**